docs/concepts/cluster-administration/device-plugins.md

```diff
@@ -0,0 +1,88 @@
+---
+approvers:
+title: Device Plugins
+---
+
+* TOC
+{:toc}
+
+__Disclaimer__: Device plugins are in alpha. Its contents may change rapidly.
+
+Starting from 1.8 release, Kubernetes provides a [device plugin framework](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/device-plugin.md)
+for vendors to advertise their resources to Kubelet without changing Kubernetes core code.
+Instead of writing custom Kubernetes code, vendors can implement a device plugin that can
+be deployed as a DaemonSet or in bare metal mode. The targeted devices include GPUs,
+High-performance NICs, FPGAs, InfiniBand, and other similar computing resources
+that may require vendor specific initialization and setup.
+
+## Overview
+The 1.8 Kubernetes release supports device plugin as an alpha feature that
+is gated by DevicePlugins feature gate and is disabled by default.
+When the DevicePlugins feature is enabled, Kubelet will export a `Registration` gRPC service:
+```gRPC
+service Registration {
+	rpc Register(RegisterRequest) returns (Empty) {}
+}
+```
+A device plugin can register itself to Kubelet through this gRPC service
+During the registration, the device plugin needs to send
+  * The name of their unix socket
+  * The API version against which they were built
+  * The `ResourceName` they want to advertise. Here `ResourceName` needs to follow
+    the [extended resource naming scheme](https://github.com/kubernetes/kubernetes/pull/48922) as `vendor-domain/resource`.
+    E.g., Nvidia GPUs are advertised as `nvidia.com/gpu`
+
+Following a successful registration, the device plugin will send Kubelet the
+list of devices it manages, and Kubelet will be in charge of advertising those
+resources to the API server as part of the Kubelet node status update.
+E.g., after a device plugin registers `vendor-domain/foo` with Kubelet
+and reports two healthy devices on a node, the node status will be updated
+to advertise 2 `vendor-domain/foo`.
+Devices can then be selected using the same process as for OIRs in the pod spec.
+Currently, extended resources are only uspported as integer resources and expect
+to always have limits == requests in container resource Spec.
+
+## Device Plugin Implementation
+
+The general workflow of a device plugin includes the following parts:
+* Initialization. During this phase, a device plugin performs vendor specific initialization and setup to make sure their devices are in ready state.
```

Comment:
a device plugin -> the device plugin

their devices -> the devices

Reply:
done.